source/fe/mapping.cc

```diff
@@ -52,6 +52,32 @@ Mapping<dim, spacedim>::get_vertices(
 
 
 
+template <int dim, int spacedim>
+boost::container::small_vector<Point<spacedim>,
+                               GeometryInfo<dim>::vertices_per_face>
+Mapping<dim, spacedim>::get_vertices(
+  const typename Triangulation<dim, spacedim>::cell_iterator &cell,
+  const unsigned int                                          face_no) const
+{
+  boost::container::small_vector<Point<spacedim>,
+                                 GeometryInfo<dim>::vertices_per_face>
+    face_vertices;
+
+  const auto  face_orientation = cell->combined_face_orientation(face_no);
+  const auto &cell_vertices    = get_vertices(cell);
+
+  for (unsigned int v = 0; v < GeometryInfo<dim>::vertices_per_face; ++v)
+    {
+      face_vertices.push_back(
+        cell_vertices[cell->reference_cell().face_to_cell_vertices(
+          face_no, v, face_orientation)]);
+    }
```

Comment:
```suggestion
  const auto reference_cell = cell->reference_cell();
  for (unsigned int v : reference_cell->face_reference_cell(face_no).vertex_indices())
    {
      face_vertices.push_back(
        cell_vertices[reference_cell.face_to_cell_vertices(face_no, v, face_orientation)]);
    }
```
Since this is in `Mapping` we should avoid using `GeometryInfo` to make this work with non-hypercube elements.

Reply:
@drwells Does this mean, ```get_vertices(cell)``` has to be updated as well? It also relies on ```GeometryInfo<dim>::vertices_per_cell```.

Reply:
Thanks for clarification. I didn't know the difference between ```boost::small_vector``` with ```std::array```.  I updated the code as suggested.